Allow using podman by passing in runtime var for systems without docker installed.

pweil- opened the pull request:
```
$ make docker-build-all
podman build -f Dockerfile.reporting-operator -t quay.io/openshift/origin-metering-reporting-operator:4.6 /home/pweil/codebase/metering-operator/src/github.com/kube-reporting/metering-operator
STEP 1: FROM openshift/origin-release:golang-1.13 AS build
Getting image source signatures
Copying blob d42565390f9b done  
Copying blob 524b0c1e57f8 done  
Copying config 2e08ed6863 done  
Writing manifest to image destination
Storing signatures
STEP 2: COPY . /go/src/github.com/kube-reporting/metering-operator
--> 64fcfcbbf54
STEP 3: WORKDIR /go/src/github.com/kube-reporting/metering-operator
--> 467c435f146
STEP 4: ENV GOCACHE='/tmp'
--> e3343eb54a8
STEP 5: RUN make reporting-operator-bin RUN_UPDATE_CODEGEN=false CHECK_GO_FILES=false
make build-reporting-operator
make[1]: Entering directory `/go/src/github.com/kube-reporting/metering-operator'
mkdir -p bin/
CGO_ENABLED=0 GOOS="linux" go build -mod=vendor -ldflags '-extldflags "-static"' -o bin/reporting-operator github.com/kube-reporting/metering-operator/cmd/reporting-operator
make[1]: Leaving directory `/go/src/github.com/kube-reporting/metering-operator'
--> 1aed0223324
STEP 6: FROM centos:7
Getting image source signatures
Copying blob e6a50b627bcb done  
Copying config 8114e7c186 done  
Writing manifest to image destination
Storing signatures
STEP 7: RUN yum install --setopt=skip_missing_names_on_install=False -y         ca-certificates bash
Loaded plugins: fastestmirror, ovl
Determining fastest mirrors
 * base: mirror.atl.genesisadaptive.com
 * extras: mirror.steadfastnet.com
 * updates: mirrors.usinternet.com
base                                                                                                                              | 3.6 kB  00:00:00     
extras                                                                                                                            | 2.9 kB  00:00:00     
updates                                                                                                                           | 2.9 kB  00:00:00     
(1/4): base/7/x86_64/group_gz                                                                                                     | 153 kB  00:00:00     
(2/4): extras/7/x86_64/primary_db                                                                                                 | 205 kB  00:00:00     
(3/4): base/7/x86_64/primary_db                                                                                                   | 6.1 MB  00:00:00     
(4/4): updates/7/x86_64/primary_db                                                                                                | 3.0 MB  00:00:00     
Resolving Dependencies
--> Running transaction check
---> Package bash.x86_64 0:4.2.46-31.el7 will be updated
---> Package bash.x86_64 0:4.2.46-34.el7 will be an update
---> Package ca-certificates.noarch 0:2018.2.22-70.0.el7_5 will be updated
---> Package ca-certificates.noarch 0:2020.2.41-70.0.el7_8 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

=========================================================================================================================================================
 Package                                Arch                          Version                                       Repository                      Size
=========================================================================================================================================================
Updating:
 bash                                   x86_64                        4.2.46-34.el7                                 base                           1.0 M
 ca-certificates                        noarch                        2020.2.41-70.0.el7_8                          updates                        382 k

Transaction Summary
=========================================================================================================================================================
Upgrade  2 Packages

Total download size: 1.4 M
Downloading packages:
Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
warning: /var/cache/yum/x86_64/7/updates/packages/ca-certificates-2020.2.41-70.0.el7_8.noarch.rpm: Header V3 RSA/SHA256 Signature, key ID f4a80eb5: NOKEY
Public key for ca-certificates-2020.2.41-70.0.el7_8.noarch.rpm is not installed
(1/2): ca-certificates-2020.2.41-70.0.el7_8.noarch.rpm                                                                            | 382 kB  00:00:00     
Public key for bash-4.2.46-34.el7.x86_64.rpm is not installed 77% [========================================            ]  0.0 B/s | 1.1 MB  --:--:-- ETA 
(2/2): bash-4.2.46-34.el7.x86_64.rpm                                                                                              | 1.0 MB  00:00:00     
---------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                    2.1 MB/s | 1.4 MB  00:00:00     
Retrieving key from file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
Importing GPG key 0xF4A80EB5:
 Userid     : "CentOS-7 Key (CentOS 7 Official Signing Key) <security@centos.org>"
 Fingerprint: 6341 ab27 53d7 8a78 a7c2 7bb1 24c6 a8a7 f4a8 0eb5
 Package    : centos-release-7-6.1810.2.el7.centos.x86_64 (@CentOS)
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : bash-4.2.46-34.el7.x86_64                                                                                                             1/4 
  Updating   : ca-certificates-2020.2.41-70.0.el7_8.noarch                                                                                           2/4 
  Cleanup    : ca-certificates-2018.2.22-70.0.el7_5.noarch                                                                                           3/4 
  Cleanup    : bash-4.2.46-31.el7.x86_64                                                                                                             4/4 
  Verifying  : ca-certificates-2020.2.41-70.0.el7_8.noarch                                                                                           1/4 
  Verifying  : bash-4.2.46-34.el7.x86_64                                                                                                             2/4 
  Verifying  : ca-certificates-2018.2.22-70.0.el7_5.noarch                                                                                           3/4 
  Verifying  : bash-4.2.46-31.el7.x86_64                                                                                                             4/4 

Updated:
  bash.x86_64 0:4.2.46-34.el7                                        ca-certificates.noarch 0:2020.2.41-70.0.el7_8                                       

Complete!
--> 1349701d665
STEP 8: COPY --from=build /go/src/github.com/kube-reporting/metering-operator/bin/reporting-operator /usr/local/bin/reporting-operator
--> 473b964526b
STEP 9: LABEL io.k8s.display-name="OpenShift metering-reporting-operator"       io.k8s.description="This is a component of OpenShift Container Platform and manages collecting data from monitoring and running reports."       io.openshift.tags="openshift"       maintainer="sd-operator-metering@redhat.com"
--> 8ea00c68195
STEP 10: ENTRYPOINT ["reporting-operator"]
--> 55b7d1ca836
STEP 11: CMD ["start"]
--> 4a63c3b429e
STEP 12: USER 3001
STEP 13: COMMIT quay.io/openshift/origin-metering-reporting-operator:4.6
--> b20953bd7b5
b20953bd7b5c34efc0adfb806582b82b99e5040d90755a79c195113bf07ca21c
podman build -f Dockerfile.metering-ansible-operator.okd -t quay.io/openshift/origin-metering-ansible-operator:4.6 /home/pweil/codebase/metering-operator/src/github.com/kube-reporting/metering-operator
STEP 1: FROM quay.io/openshift/origin-metering-helm:latest AS helm
Getting image source signatures
Copying blob 975d704d6ce8 done  
Copying blob ab5ef0e58194 done  
Copying config 766ba6cdc0 done  
Writing manifest to image destination
Storing signatures
--> 766ba6cdc09
STEP 2: FROM openshift/origin-cli:latest AS cli
Getting image source signatures
Copying blob c2340472a0fa done  
Copying blob b2d2704dda6c done  
Copying blob a2c10be042b9 done  
Copying blob 6e55351c18ff done  
Copying blob cc268c59b452 done  
Copying config 7ebb6be011 done  
Writing manifest to image destination
Storing signatures
--> 7ebb6be0111
STEP 3: FROM quay.io/openshift/origin-ansible-operator:4.6
Getting image source signatures
Copying blob 78afc5364ad2 done  
Copying blob 5de17b44177d done  
Copying blob 58e1deb9693d done  
Copying config c0554484fd done  
Writing manifest to image destination
Storing signatures
STEP 4: USER root
--> edaa28466b3
STEP 5: RUN set -x; INSTALL_PKGS="curl bash ca-certificates less which inotify-tools openssl"     && yum clean all && rm -rf /var/cache/yum/*     && yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm      && yum install --setopt=skip_missing_names_on_install=False -y         $INSTALL_PKGS      && yum clean all     && rm -rf /var/cache/yum
+ INSTALL_PKGS='curl bash ca-certificates less which inotify-tools openssl'
+ yum clean all
Updating Subscription Management repositories.
Unable to read consumer identity
/usr/local/lib/python3.6/site-packages/dateutil/parser/_parser.py:70: UnicodeWarning: decode() called on unicode string, see https://bugzilla.redhat.com/show_bug.cgi?id=1693751
  instream = instream.decode()

This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
0 files removed
+ rm -rf '/var/cache/yum/*'
+ yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
Updating Subscription Management repositories.
Unable to read consumer identity
/usr/local/lib/python3.6/site-packages/dateutil/parser/_parser.py:70: UnicodeWarning: decode() called on unicode string, see https://bugzilla.redhat.com/show_bug.cgi?id=1693751
  instream = instream.decode()

This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
Red Hat Universal Base Image 8 (RPMs) - BaseOS                                                                           1.8 MB/s | 766 kB     00:00    
Red Hat Universal Base Image 8 (RPMs) - AppStream                                                                        8.7 MB/s | 3.9 MB     00:00    
Red Hat Universal Base Image 8 (RPMs) - CodeReady Builder                                                                 29 kB/s |  11 kB     00:00    
epel-release-latest-8.noarch.rpm                                                                                          69 kB/s |  22 kB     00:00    
Dependencies resolved.
=========================================================================================================================================================
 Package                                Architecture                     Version                            Repository                              Size
=========================================================================================================================================================
Installing:
 epel-release                           noarch                           8-8.el8                            @commandline                            22 k

Transaction Summary
=========================================================================================================================================================
Install  1 Package

Total size: 22 k
Installed size: 32 k
Downloading Packages:
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                 1/1 
  Installing       : epel-release-8-8.el8.noarch                                                                                                     1/1 
  Running scriptlet: epel-release-8-8.el8.noarch                                                                                                     1/1 
  Verifying        : epel-release-8-8.el8.noarch                                                                                                     1/1 
Installed products updated.

Installed:
  epel-release-8-8.el8.noarch                                                                                                                            

Complete!
+ yum install --setopt=skip_missing_names_on_install=False -y curl bash ca-certificates less which inotify-tools openssl
Main config did not have a skip_missing_names_on_install attr. before setopt
Updating Subscription Management repositories.
Unable to read consumer identity
/usr/local/lib/python3.6/site-packages/dateutil/parser/_parser.py:70: UnicodeWarning: decode() called on unicode string, see https://bugzilla.redhat.com/show_bug.cgi?id=1693751
  instream = instream.decode()

This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
Main config did not have a skip_missing_names_on_install attr. before setopt
Extra Packages for Enterprise Linux Modular 8 - x86_64                                                                    54 kB/s |  82 kB     00:01    
Extra Packages for Enterprise Linux 8 - x86_64                                                                           8.8 MB/s | 7.4 MB     00:00    
Package curl-7.61.1-12.el8.x86_64 is already installed.
Package bash-4.4.19-10.el8.x86_64 is already installed.
Package ca-certificates-2019.2.32-80.0.el8_1.noarch is already installed.
Package which-2.21-12.el8.x86_64 is already installed.
Package inotify-tools-3.14-16.fc30.x86_64 is already installed.
Dependencies resolved.
=========================================================================================================================================================
 Package                               Architecture                   Version                                 Repository                            Size
=========================================================================================================================================================
Installing:
 less                                  x86_64                         530-1.el8                               ubi-8-baseos                         164 k
 openssl                               x86_64                         1:1.1.1c-15.el8                         ubi-8-baseos                         697 k
Upgrading:
 inotify-tools                         x86_64                         3.14-19.el8                             epel                                  57 k

Transaction Summary
=========================================================================================================================================================
Install  2 Packages
Upgrade  1 Package

Total download size: 918 k
Downloading Packages:
(1/3): less-530-1.el8.x86_64.rpm                                                                                         882 kB/s | 164 kB     00:00    
(2/3): openssl-1.1.1c-15.el8.x86_64.rpm                                                                                  3.0 MB/s | 697 kB     00:00    
(3/3): inotify-tools-3.14-19.el8.x86_64.rpm                                                                              241 kB/s |  57 kB     00:00    
---------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                    1.2 MB/s | 918 kB     00:00     
warning: /var/cache/dnf/epel-fafd94c310c51e1e/packages/inotify-tools-3.14-19.el8.x86_64.rpm: Header V3 RSA/SHA256 Signature, key ID 2f86d6a1: NOKEY
Extra Packages for Enterprise Linux 8 - x86_64                                                                           1.6 MB/s | 1.6 kB     00:00    
Importing GPG key 0x2F86D6A1:
 Userid     : "Fedora EPEL (8) <epel@fedoraproject.org>"
 Fingerprint: 94E2 79EB 8D8F 25B2 1810 ADF1 21EA 45AB 2F86 D6A1
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-8
Key imported successfully
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                 1/1 
  Upgrading        : inotify-tools-3.14-19.el8.x86_64                                                                                                1/4 
  Installing       : openssl-1:1.1.1c-15.el8.x86_64                                                                                                  2/4 
  Installing       : less-530-1.el8.x86_64                                                                                                           3/4 
  Cleanup          : inotify-tools-3.14-16.fc30.x86_64                                                                                               4/4 
  Running scriptlet: inotify-tools-3.14-16.fc30.x86_64                                                                                               4/4 
  Verifying        : less-530-1.el8.x86_64                                                                                                           1/4 
  Verifying        : openssl-1:1.1.1c-15.el8.x86_64                                                                                                  2/4 
  Verifying        : inotify-tools-3.14-19.el8.x86_64                                                                                                3/4 
  Verifying        : inotify-tools-3.14-16.fc30.x86_64                                                                                               4/4 
Installed products updated.

Upgraded:
  inotify-tools-3.14-19.el8.x86_64                                                                                                                       

Installed:
  less-530-1.el8.x86_64                                                  openssl-1:1.1.1c-15.el8.x86_64                                                 

Complete!
+ yum clean all
Updating Subscription Management repositories.
Unable to read consumer identity
/usr/local/lib/python3.6/site-packages/dateutil/parser/_parser.py:70: UnicodeWarning: decode() called on unicode string, see https://bugzilla.redhat.com/show_bug.cgi?id=1693751
  instream = instream.decode()

This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
43 files removed
+ rm -rf /var/cache/yum
--> f982026ff13
STEP 6: ENV TINI_VERSION v0.18.0
--> 2367b3468f3
STEP 7: ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
--> 14a4d36b727
STEP 8: RUN chmod +x /tini
--> f4bbfb610fe
STEP 9: COPY --from=helm /usr/local/bin/helm /usr/local/bin/helm
--> 4ba908d2bb5
STEP 10: COPY --from=cli /usr/bin/oc /usr/bin/oc
--> 12194692fb6
STEP 11: RUN ln -f -s /usr/bin/oc /usr/bin/kubectl
--> 8b7873b26f5
STEP 12: RUN ln -f -s /tini /usr/bin/tini
--> 400a83a14d5
STEP 13: RUN pip install --no-cache-dir --upgrade ansible~=2.9 openshift botocore boto3 cryptography netaddr
Collecting ansible~=2.9
  Downloading ansible-2.9.11.tar.gz (14.2 MB)
     |████████████████████████████████| 14.2 MB 3.2 MB/s 
Collecting openshift
  Downloading openshift-0.11.2.tar.gz (19 kB)
Collecting botocore
  Downloading botocore-1.17.24-py2.py3-none-any.whl (6.4 MB)
     |████████████████████████████████| 6.4 MB 25.8 MB/s 
Collecting boto3
  Downloading boto3-1.14.24-py2.py3-none-any.whl (128 kB)
     |████████████████████████████████| 128 kB 80.1 MB/s 
Collecting cryptography
  Downloading cryptography-3.0-cp35-abi3-manylinux2010_x86_64.whl (2.7 MB)
     |████████████████████████████████| 2.7 MB 76.9 MB/s 
Collecting netaddr
  Downloading netaddr-0.8.0-py2.py3-none-any.whl (1.9 MB)
     |████████████████████████████████| 1.9 MB 77.8 MB/s 
Requirement already satisfied, skipping upgrade: jinja2 in /usr/local/lib/python3.6/site-packages (from ansible~=2.9) (2.11.2)
Requirement already satisfied, skipping upgrade: PyYAML in /usr/local/lib64/python3.6/site-packages (from ansible~=2.9) (5.3.1)
Collecting kubernetes~=11.0.0
  Downloading kubernetes-11.0.0-py3-none-any.whl (1.5 MB)
     |████████████████████████████████| 1.5 MB 54.6 MB/s 
Requirement already satisfied, skipping upgrade: python-string-utils in /usr/local/lib/python3.6/site-packages (from openshift) (1.0.0)
Requirement already satisfied, skipping upgrade: ruamel.yaml>=0.15 in /usr/local/lib/python3.6/site-packages (from openshift) (0.16.10)
Requirement already satisfied, skipping upgrade: six in /usr/local/lib/python3.6/site-packages (from openshift) (1.15.0)
Requirement already satisfied, skipping upgrade: python-dateutil<3.0.0,>=2.1 in /usr/local/lib/python3.6/site-packages (from botocore) (2.8.1)
Requirement already satisfied, skipping upgrade: jmespath<1.0.0,>=0.7.1 in /usr/local/lib/python3.6/site-packages (from botocore) (0.10.0)
Requirement already satisfied, skipping upgrade: urllib3<1.26,>=1.20; python_version != "3.4" in /usr/local/lib/python3.6/site-packages (from botocore) (1.25.9)
Collecting docutils<0.16,>=0.10
  Downloading docutils-0.15.2-py3-none-any.whl (547 kB)
     |████████████████████████████████| 547 kB 79.3 MB/s 
Collecting s3transfer<0.4.0,>=0.3.0
  Downloading s3transfer-0.3.3-py2.py3-none-any.whl (69 kB)
     |████████████████████████████████| 69 kB 66.6 MB/s 
Requirement already satisfied, skipping upgrade: cffi!=1.11.3,>=1.8 in /usr/local/lib64/python3.6/site-packages (from cryptography) (1.14.0)
Collecting importlib-resources; python_version < "3.7"
  Downloading importlib_resources-3.0.0-py2.py3-none-any.whl (23 kB)
Requirement already satisfied, skipping upgrade: MarkupSafe>=0.23 in /usr/local/lib64/python3.6/site-packages (from jinja2->ansible~=2.9) (1.1.1)
Requirement already satisfied, skipping upgrade: certifi>=14.05.14 in /usr/local/lib/python3.6/site-packages (from kubernetes~=11.0.0->openshift) (2020.4.5.1)
Requirement already satisfied, skipping upgrade: requests-oauthlib in /usr/local/lib/python3.6/site-packages (from kubernetes~=11.0.0->openshift) (1.3.0)
Requirement already satisfied, skipping upgrade: google-auth>=1.0.1 in /usr/local/lib/python3.6/site-packages (from kubernetes~=11.0.0->openshift) (1.16.0)
Requirement already satisfied, skipping upgrade: setuptools>=21.0.0 in /usr/local/lib/python3.6/site-packages (from kubernetes~=11.0.0->openshift) (47.1.1)
Requirement already satisfied, skipping upgrade: requests in /usr/local/lib/python3.6/site-packages (from kubernetes~=11.0.0->openshift) (2.23.0)
Requirement already satisfied, skipping upgrade: websocket-client!=0.40.0,!=0.41.*,!=0.42.*,>=0.32.0 in /usr/local/lib/python3.6/site-packages (from kubernetes~=11.0.0->openshift) (0.57.0)
Requirement already satisfied, skipping upgrade: ruamel.yaml.clib>=0.1.2; platform_python_implementation == "CPython" and python_version < "3.9" in /usr/local/lib64/python3.6/site-packages (from ruamel.yaml>=0.15->openshift) (0.2.0)
Requirement already satisfied, skipping upgrade: pycparser in /usr/local/lib/python3.6/site-packages (from cffi!=1.11.3,>=1.8->cryptography) (2.20)
Collecting zipp>=0.4; python_version < "3.8"
  Downloading zipp-3.1.0-py3-none-any.whl (4.9 kB)
Requirement already satisfied, skipping upgrade: oauthlib>=3.0.0 in /usr/local/lib/python3.6/site-packages (from requests-oauthlib->kubernetes~=11.0.0->openshift) (3.1.0)
Requirement already satisfied, skipping upgrade: pyasn1-modules>=0.2.1 in /usr/local/lib/python3.6/site-packages (from google-auth>=1.0.1->kubernetes~=11.0.0->openshift) (0.2.8)
Requirement already satisfied, skipping upgrade: rsa<4.1,>=3.1.4 in /usr/local/lib/python3.6/site-packages (from google-auth>=1.0.1->kubernetes~=11.0.0->openshift) (4.0)
Requirement already satisfied, skipping upgrade: cachetools<5.0,>=2.0.0 in /usr/local/lib/python3.6/site-packages (from google-auth>=1.0.1->kubernetes~=11.0.0->openshift) (4.1.0)
Requirement already satisfied, skipping upgrade: chardet<4,>=3.0.2 in /usr/local/lib/python3.6/site-packages (from requests->kubernetes~=11.0.0->openshift) (3.0.4)
Requirement already satisfied, skipping upgrade: idna<3,>=2.5 in /usr/local/lib/python3.6/site-packages (from requests->kubernetes~=11.0.0->openshift) (2.9)
Requirement already satisfied, skipping upgrade: pyasn1<0.5.0,>=0.4.6 in /usr/local/lib/python3.6/site-packages (from pyasn1-modules>=0.2.1->google-auth>=1.0.1->kubernetes~=11.0.0->openshift) (0.4.8)
Using legacy setup.py install for ansible, since package 'wheel' is not installed.
Using legacy setup.py install for openshift, since package 'wheel' is not installed.
Installing collected packages: cryptography, ansible, kubernetes, openshift, docutils, botocore, s3transfer, boto3, zipp, importlib-resources, netaddr
  Attempting uninstall: cryptography
    Found existing installation: cryptography 2.9.2
    Uninstalling cryptography-2.9.2:
      Successfully uninstalled cryptography-2.9.2
  Attempting uninstall: ansible
    Found existing installation: ansible 2.9.9
    Uninstalling ansible-2.9.9:
      Successfully uninstalled ansible-2.9.9
    Running setup.py install for ansible ... done
  Attempting uninstall: kubernetes
    Found existing installation: kubernetes 10.1.0
    Uninstalling kubernetes-10.1.0:
      Successfully uninstalled kubernetes-10.1.0
  Attempting uninstall: openshift
    Found existing installation: openshift 0.10.3
    Uninstalling openshift-0.10.3:
      Successfully uninstalled openshift-0.10.3
    Running setup.py install for openshift ... done
  Attempting uninstall: docutils
    Found existing installation: docutils 0.16
    Uninstalling docutils-0.16:
      Successfully uninstalled docutils-0.16
Successfully installed ansible-2.9.11 boto3-1.14.24 botocore-1.17.24 cryptography-3.0 docutils-0.15.2 importlib-resources-3.0.0 kubernetes-11.0.0 netaddr-0.8.0 openshift-0.11.2 s3transfer-0.3.3 zipp-3.1.0
--> b2931df0ffc
STEP 14: ENV HOME /opt/ansible
--> b65703b0f20
STEP 15: ENV HELM_CHART_PATH ${HOME}/charts/openshift-metering
--> 3b849d6b018
STEP 16: COPY images/metering-ansible-operator/roles/ ${HOME}/roles/
--> 6247a172932
STEP 17: COPY images/metering-ansible-operator/watches.yaml ${HOME}/watches.yaml
--> 2ebb91ba40c
STEP 18: COPY images/metering-ansible-operator/scripts ${HOME}/scripts
--> acbd61fcca0
STEP 19: COPY images/metering-ansible-operator/ansible.cfg /etc/ansible/ansible.cfg
--> 50d72e2afae
STEP 20: COPY charts/openshift-metering ${HELM_CHART_PATH}
--> 342bc5ab70e
STEP 21: COPY manifests/deploy/openshift/olm/bundle /manifests
--> 0680e3760fe
STEP 22: USER 1001
--> 8ce0b2e604c
STEP 23: ENTRYPOINT ["tini", "--", "/usr/local/bin/ansible-operator", "exec-entrypoint", "ansible", "--watches-file", "/opt/ansible/watches.yaml"]
--> 313d0e0bc1b
STEP 24: LABEL io.k8s.display-name="OpenShift metering-ansible-operator"       io.k8s.description="This is a component of OpenShift Container Platform and manages installation and configuration of all other metering components."       io.openshift.tags="openshift"       com.redhat.delivery.appregistry=true       maintainer="sd-operator-metering@redhat.com"
STEP 25: COMMIT quay.io/openshift/origin-metering-ansible-operator:4.6
--> badbc21857d
badbc21857ddce118e84cd6d4ac1b585284904da62132758116999fc6066097c
[pweil@pweil-mac metering-operator]$ 
```

pweil- commented:
/assign @bentito

pruan-rht commented on Makefile:
```diff
@@ -11,7 +11,15 @@ DEPLOY_METERING_PKG := $(GO_PKG)/cmd/deploy-metering
 
+# check if podman is installed and if so, use it.  Otherwise try to use docker.
+ifeq (, $(shell which podman))
+$(info podman not found, using docker for build command)
```
@pweil- indentation?

pruan-rht commented on Makefile:
```diff
@@ -11,7 +11,15 @@ DEPLOY_METERING_PKG := $(GO_PKG)/cmd/deploy-metering
+$(info podman not found, using docker for build command)
 DOCKER_BUILD_CMD = docker build
+else
+$(info using podman for build command)
```
@pweil- ditto

pweil- commented:
/hold 

Someone with Mac should test this before we merge in addition to the CI tests :smile:

pweil- commented:
/retest

pweil- commented:
/test metering-upgrade-aws

pweil- commented:
@bentito clean CI on this now.  Please give it a go on your mac to confirm you're still picking up the right executable and this is ready for review.

bentito commented:
@pweil- as-is:
```bash
metering-operator pull/1323 $ make docker-build-all
using podman for build command
podman build -f Dockerfile.reporting-operator -t quay.io/openshift/origin-metering-reporting-operator:4.6 /Users/btofel/workspace/metering-operator
Error: could not get runtime: dial unix /run/podman/io.podman: connect: no such file or directory
make: *** [reporting-operator-docker-build] Error 125
```
Doesn't seem like it takes an env var for whether to use podman or docker yet?

pweil- commented:
> Doesn't seem like it takes an env var for whether to use podman or docker yet?

It isn't looking for an env var, it is checking for the existence on the path.  I'll have to break out the mac and see what's up I guess.  Thanks for giving it a quick run.

pweil- commented:
oh, no, wait.  Podman actually was on your path so that part is working.  Podman just cannot connect.

pweil- commented:
removed the info output, it was just confusing.  

turns out, since we copy and run this `Makefile` in containers during the container build process you can even see the info message of "using docker" even on a system that is building the container with `podman` since obviously `podman` is not in the running container's path.  

Also updated the rest of the file to remove unnecessary references to docker within the file but not in any of the target names since I am ignorant on where they may be used in other scripts.

Will break out the mac and test there at some point this week.  Thanks to @bentito for some initial testing feedback.

timflannagan1 commented:
/retest

timflannagan1 commented:
Failing due to known many-to-many errors when grabbing PVC-related metrics from Prometheus.
/override ci/prow/metering-upgrade-aws

pweil- commented:
Was discussing image builds with one of our registry folks and they mentioned that the oci compliant manifests that podman build produces are not handled nicely by our internal registry yet without setting some flags for generating the right schema.  I think that's enough to make me think this is probably closeable.

For folks who are developing on a system without docker we can look at the env var route Brett mentioned before vs preferring podman if available.

timflannagan1 commented on Makefile:
```diff
@@ -11,15 +11,23 @@ DEPLOY_METERING_PKG := $(GO_PKG)/cmd/deploy-metering
 VERIFY_FILE_PATHS := cmd pkg test manifests
 
-DOCKER_BUILD_CMD = docker build
+# Prefer podman, if not on the path use docker.  Suppress errors so we don't see any
+# potentially misleading messages like "which: no podman in ..." in the output.
+CONTAINER_RUNTIME_CMD = podman
+ifeq (, $(shell which podman 2> /dev/null))
+    CONTAINER_RUNTIME_CMD = docker
+endif
```
Can we just remove this block and instead just default `CONTAINER_RUNTIME_CMD = docker` with the intention that you would override it when running a make target, e.g. `make verify-docker CONTAINER_RUNTIME_CMD=podman`

timflannagan1 commented:
I'm down to re-open if we just also default to docker and remove all hardcoded references to docker with `$(CONTAINER_RUNTIME_CMD)` or `$(IMAGE_BUILD_CMD)` as that's better than what we have.

pweil- commented:
Yeah, I can change it around to do that.

pweil- commented:
updated:

```
$ make docker-build-all
docker build -f Dockerfile.reporting-operator -t quay.io/openshift/origin-metering-reporting-operator:4.6 /home/pweil/codebase/metering-operator/src/github.com/kube-reporting/metering-operator
/bin/bash: docker: command not found
make: *** [Makefile:93: reporting-operator-docker-build] Error 127


[pweil@paperweight metering-operator]$ make docker-build-all CONTAINER_RUNTIME_CMD=podman
podman build -f Dockerfile.reporting-operator -t quay.io/openshift/origin-metering-reporting-operator:4.6 /home/pweil/codebase/metering-operator/src/github.com/kube-reporting/metering-operator
STEP 1: FROM openshift/origin-release:golang-1.13 AS build
STEP 2: COPY . /go/src/github.com/kube-reporting/metering-operator
```

pweil- commented:
/hold cancel

timflannagan1 commented:
/lgtm
